### DISCUSSION TITLE
MacOS Caps Lock as CTRL/ESC Using Better Touch Tool

### DISCUSSION THREAD

**Discussion by TomRobinsonIO** [ghostty-org/ghostty#4083] [Q&A]
Hello, I have a somewhat unique workflow where I have my Caps Lock key mapped to Control in MacOS. I then use the _Key Sequence_ feature of Better Touch Tool to effectively make the key behave so that if it is tapped and quickly released, it will send an Escape. However if it is held, BTT ignores the key and it acts as Control. This has worked well for me in many other terminal emulators but does not appear to play nice with Ghostty. 

From the _Keyboard_ section of the Terminal Inspector, I can see the events for pressing the dedicated Escape key:

`Release: escape`
`Press: escape`

However, when using the Caps Lock key with the BTT _Key Sequence_:

`Release: left_control`
`Release: escape`
`Press: Ctrl+left_control`
`Release: right_control`
`Press: Ctrl+right_control`

Any advice would be appreciated and please let me know if more information needs to be provided. Thank you!

**Comment by faumahesh** [comment]
Similar issue - Ghosty appears to ignore the CAPS to Control mapping setup in OSX.
<img width="650" height="455" alt="SCR-20260308-ktgk" src="https://github.com/user-attachments/assets/ee7a18d0-c4e3-456d-a979-ed2ff9f902c5" />

**Reply by karl-gustav** [reply]
I have used caps-lock as ctrl in ghostty(MacOS) for a long time. Works like a charm 🤔

**Reply by faumahesh** [reply]
Ack! I went back and double checked. You were right! I had the wrong keyboard selected. In my screenshot above, I had "USB Receiver," whereas I have since switched to a Magic Keyboard. It does indeed work. - Thanks!

<img width="650" height="453" alt="SCR-20260308-lmyj" src="https://github.com/user-attachments/assets/a7b806dc-ecc0-4aae-aa33-dcfdc9931a6b" />
